In [ ]:
import pandas as pd
countries_df = pd.DataFrame({
    'country': ['Англия', 'Канада', 'США', 'Россия', 'Украина', 'Беларусь', 'Казахстан'],
    'population': [56.29, 38.05, 322.28, 146.24, 45.5, 9.5, 17.04],
    'square': [133396, 9984670, 9826630, 17125191, 603628, 207600, 2724902]
})

countries_df.to_csv('data/countries.csv', index=False, sep=';')

countries_data = pd.read_csv('data/countries.csv', sep=';')
# display(countries_data)

data = pd.read_csv('https://raw.githubusercontent.com/esabunor/MLWorkspace/master/melb_data.csv')
# display(data)

import pandas as pd

melb_data = pd.read_csv('data/melb_data.csv', sep=',')
# display(round(melb_data.iloc[3521, 14]/melb_data.iloc[1690, 14])) Восколько раз объект больше другого
# display(melb_data.head(7)) # First 5 elements
# display(melb_data.tail(7)) # Last 5 elements
# display(melb_data.shape) # узнать размер таблицы — количество строк и количество столбцов

melb_data['Car'] = melb_data['Car'].astype('int64')
melb_data['Bedroom'] = melb_data['Bedroom'].astype('int64')
melb_data['Bathroom'] = melb_data['Bathroom'].astype('int64')
melb_data['Propertycount'] = melb_data['Propertycount'].astype('int64')
melb_data['YearBuilt'] = melb_data['YearBuilt'].astype('int64')
display(melb_data.info())

# display(melb_data.describe()) # метод describe() возвращает DataFrame, в котором указаны: 

# количество непустых строк (count); 
# количество уникальных значений (unique); 
# самое частое значение — мода —  (top); 
# частота — объём использования — этого значения (freq) для каждого столбца типа object исходной таблицы.

In [56]:
melb_data.describe(include=['object']) # С помощью параметра include можно указать 
# тип данных, для которого нужно вывести описательную информацию

Suburb           Address   Type Method SellerG        Date  \
count       13580             13580  13580  13580   13580       13580   
unique        314             13378      3      5     268          58   
top     Reservoir  36 Aberfeldie St      h      S  Nelson  27/05/2017   
freq          359                 3   9449   9022    1565         473   

       CouncilArea             Regionname         Coordinates  
count        12211                  13580               13580  
unique          33                      8               13097  
top       Moreland  Southern Metropolitan  -37.8361, 144.9966  
freq          1163                   4695                  12

In [57]:
display(13580-12211)

1369

Столбец Suburb (пригород)

Наибольшее количество проданных объектов (359) находилось в пригороде Reservoir.
Столбец Type (тип здания)

Самый популярный тип дома — h (дом, вилла, коттедж).
Столбец SellerG (риелтор)

В наших данных нам известно о 268 различных риэлторских компаниях, однако самой главной «акулой» в этом бизнесе является компания Nelson — они продали 1 565 различных домов.
Столбец Date (дата продажи)

В нашей таблице содержится информация за 58 дней, при этом наибольшее число продаж (473) пришлось на 27 мая 2017 года.

In [58]:
melb_data['Regionname'].value_counts() # в качестве индексов выступают уникальные категории столбца, 
# а значениями — соответствующая им частота

Regionname
Southern Metropolitan         4695
Northern Metropolitan         3890
Western Metropolitan          2948
Eastern Metropolitan          1471
South-Eastern Metropolitan     450
Eastern Victoria                53
Northern Victoria               41
Western Victoria                32
Name: count, dtype: int64

In [62]:
melb_data.describe(include=['object'])

Suburb           Address   Type Method SellerG        Date  \
count       13580             13580  13580  13580   13580       13580   
unique        314             13378      3      5     268          58   
top     Reservoir  36 Aberfeldie St      h      S  Nelson  27/05/2017   
freq          359                 3   9449   9022    1565         473   

       CouncilArea             Regionname         Coordinates  
count        12211                  13580               13580  
unique          33                      8               13097  
top       Moreland  Southern Metropolitan  -37.8361, 144.9966  
freq          1163                   4695                  12

In [63]:
melb_data['Price'].value_counts()

Price
1100000.0    113
1300000.0    109
650000.0     109
800000.0     109
600000.0     104
            ... 
1928000.0      1
2236000.0      1
601500.0       1
550500.0       1
1323000.0      1
Name: count, Length: 2204, dtype: int64